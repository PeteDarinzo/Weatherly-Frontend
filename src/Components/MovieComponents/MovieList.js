import React, { useEffect, useState } from "react";
import MovieCard from "./MovieCard";
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';
import { useSelector, useDispatch } from "react-redux";
import { fetchTitlesFromAPI } from "../../Actions/actions";
import Grid from '@mui/material/Grid';
import { makeStyles } from "@mui/styles";
import { Typography } from "@mui/material";
import { ConstructionOutlined } from "@mui/icons-material";

const useStyles = makeStyles({
  container: {
    padding: 40
  }
});

const MovieList = () => {

  const classes = useStyles();

  const titles = useSelector(store => store.titles);

  if (!titles.length) return (<b>No movies to show!</b>);

  return (
    <Container className={classes.container}>
      <Typography mb={4} variant="h2">Your Movies:</Typography>
      <Grid container spacing={4} >
        {titles.map(m => {
          return (
            <Grid item xs={12} sm={4} md={3} lg={2} key={m.id}>
              <MovieCard id={m.id} title={m.title} img={m.posterUrl} plot={m.plot} />
            </Grid>);
        })}
      </Grid>
    </Container>
  );
}

export default MovieList;