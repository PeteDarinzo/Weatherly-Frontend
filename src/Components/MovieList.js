import React, { useEffect, useState } from "react";
import MovieCard from "./MovieCard";
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';
import { useSelector, useDispatch } from "react-redux";
import { fetchTitlesFromAPI } from "../Actions/actions";

const MovieList = () => {

  const titles = useSelector(store => store.titles);

  const dispatch = useDispatch();

  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    async function loadTitles() {
      await dispatch(fetchTitlesFromAPI("peter"));
      setIsLoading(false);
    }

    if (isLoading) loadTitles();

  }, [dispatch, isLoading]);

  if (isLoading) return (<b>Loading...</b>);

  return (
    <Container maxWidth="sm">
      <Stack spacing={2}>
        {titles.map(m => <MovieCard id={m.id} title={m.title} img={m.posterUrl} plot={m.plot} key={m.id} />)}
      </Stack>
    </Container>
  );
}

export default MovieList;