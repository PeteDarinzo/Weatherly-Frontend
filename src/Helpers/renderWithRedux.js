import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}