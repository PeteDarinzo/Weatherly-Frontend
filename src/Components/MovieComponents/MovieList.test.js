import MovieList from "./MovieList";
import { MemoryRouter } from "react-router-dom";
import renderWithRedux from "../../Helpers/renderWithRedux";


it("renders the movie list without crashing", () => {
  renderWithRedux(
    <MemoryRouter>
      <MovieList />
    </MemoryRouter>
  );
});


it("matches the movie list snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <MovieList  />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
