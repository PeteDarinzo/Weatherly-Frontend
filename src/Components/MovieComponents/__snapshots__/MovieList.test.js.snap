// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the movie list snapshot 1`] = `
<DocumentFragment>
  <b>
    No movies to show!
  </b>
</DocumentFragment>
`;
